relay-server/src/metrics_extraction/event.rs

```diff
@@ -49,14 +49,12 @@ impl Extractable for Span {
 /// If this is a transaction event with spans, metrics will also be extracted from the spans.
 pub fn extract_metrics(
     event: &mut Event,
-    spans_extracted: bool,
     config: CombinedMetricExtractionConfig<'_>,
     max_tag_value_size: usize,
     span_extraction_sample_rate: Option<f32>,
 ) -> Vec<Bucket> {
     let mut metrics = generic::extract_metrics(event, config);
-    // If spans were already extracted for an event, we rely on span processing to extract metrics.
-    if !spans_extracted && sample(span_extraction_sample_rate.unwrap_or(1.0)) {
+    if sample(span_extraction_sample_rate.unwrap_or(1.0)) {
```

Comment:
We needed this check to avoid double extraction for transactions that were created from spans, feature that [has been removed](https://github.com/getsentry/relay/pull/3801).